test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -202,4 +222,15 @@ namespace integration_framework {
     sendYacState(my_votes);
   }
 
+  void FakePeer::sendProposal(
+      std::unique_ptr<shared_model::interface::Proposal> proposal) {
+    os_transport_->publishProposal(std::move(proposal),
+                                   {real_peer_->address()});
+  }
+
+  void FakePeer::sendBatch(
+      const std::shared_ptr<shared_model::interface::TransactionBatch> &batch) {
```

Comment:
Is there sense in passing pointer by const ref?

Reply:
The only usage of this pointer is to be passed to `propagateBatch`, which accepts it by value, so the copy still will always take place. It will be better, I think, if you accept the pointer by value here and then move it into `propagateBatch`.

Reply:
One of the suggestions for using `const&` with smart pointers is when it is only for "observing the value", so no modifications. So maybe a better option here is to show in `sendBatch` interface, that the value will be copied.

Reply:
It seems to me that this case should be the observing of the value, and that `OrderingGateTransportGrpc::propagateBatch(OrderingGateTransportGrpc::propagateBatch)` should better take a shared pointer const ref, as it is a synchronous blocking call.